adds structure test to ensure no java 7 runtime gets pulled in

balopat opened the pull request:
closes #55.

aslo commented on openjdk8/src/test/resources/structure.yaml:
```diff
@@ -8,6 +8,11 @@ commandTests:
+  setup: [[ 'chmod', '+x', '/workspace/openjdk8/src/test/workspace/test-no-java-7.bash' ]]
+  command: [ '/workspace/openjdk8/src/test/workspace/test-no-java-7.bash' ]
+  expectedError: ['WARNING.*apt does not have a stable CLI interface yet.']
+  expectedOutput: ['OK.* is not installed']
```
I'm not sure that this will catch if java 7 is installed. I believe that the test will pass if the `expectedOutput` or `expectedError` are matched at least one time.

balopat replied:
Agreed, thank you, for catching this! 

I tried it out with java 8 and it would have not caught it. 
I committed a new version that explicitly states "NOT OK. xxxx is installed" and now the test matches that as the ``excludedOutput``. I also removed ``expectedError`` as it is not really expected, just meant to be an explanatory piece - however it might just be more distracting than anything.

aslo commented:
Nice! Testing logic looks good. 

Did you consider running `apt-get update` before running the test? While building the openjdk docker container I believe we clear the apt caches, so it's possible that the `apt-cache dump`  command that you use to find java 7 packages might not be fully populated.

balopat commented:
I rewrote the check for an extensible blacklist checking test rather. 
The list of java/jre/jdk 7 packages are part of it now (maybe too much?). 
But this is much more understandable and safe. 
Thus, apt-get update is unnecessary. I also switched to ``dpkg``, which removed the annoying warning from the apt CLI.

meltsufin commented on openjdk8/src/test/workspace/test-blacklisted-pkgs.bash:
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+BLACKLIST="jvm-7-avian-jre"
+BLACKLIST="java7-runtime-headless $BLACKLIST"
```
I think it's a bit verbose in this form.
Maybe just:
```sh
BLACKLIST="\
jvm-7-avian-jre \
libjsr107cache-java \
...
"
```

balopat replied:
Done! Thank you!
